core/include/vecmem/memory/resources/terminal/malloc_resource.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "vecmem/memory/resources/memory_resource.hpp"
+
+#include <memory>
+#include <cstdlib>
+
+namespace vecmem::resource {
```

Comment:
I would not use the `resource` sub-namespace here. As we talked about it before, I foresee calling the CUDA resources `vecmem::cuda::device_memory`. Possibly `vecmem::cuda::device_memory_resource`. I don't want to call it `vecmem::cuda::resource::device_resource`...

So with that in mind I myself would call this class `vecmem::host_memory`... (With CUDA having a `vecmem::cuda::host_memory` later on.)